### PR TITLE
Add cursor rules file

### PR DESCRIPTION
**Pull request by danfinlay**
This is my current best draft of a preprompt to help development of caveat enforcers.

It will need to be kept up to date when we change interfaces. I believe I have it up to date with main now.

### CODE REVIEW

File: .cursorrules [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+So any policy that is composed of those can be assumed provided already.
+
+In the case that you need to create a custom enforcer, you can use the `CaveatEnforcer.sol` base class and write your own like this:
+
+```solidity
+// SPDX-License-Identifier: MIT AND Apache-2.0
+pragma solidity 0.8.23;
+
+import "forge-std/Test.sol";
```
Comment (hanzel98) [inline .cursorrules:63]: Not needed

File: .cursorrules [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+import { Execution, Caveat, Delegation, ModeCode } from "../../src/utils/Types.sol";
+import { Counter } from "../utils/Counter.t.sol";
+import { CaveatEnforcerBaseTest } from "./CaveatEnforcerBaseTest.t.sol";
+import { AllowedMethodsEnforcer } from "../../src/enforcers/AllowedMethodsEnforcer.sol";
+import { IDelegationManager } from "../../src/interfaces/IDelegationManager.sol";
+import { EncoderLib } from "../../src/libraries/EncoderLib.sol";
+import { ICaveatEnforcer } from "../../src/interfaces/ICaveatEnforcer.sol";
+
+contract AllowedMethodsEnforcerTest is CaveatEnforcerBaseTest {
+    using ModeLib for ModeCode;
```
Comment (hanzel98) [inline .cursorrules:77]: It might not be needed

File: .cursorrules [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+import "forge-std/Test.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ModeLib } from "@erc7579/lib/ModeLib.sol";
+import { ExecutionLib } from "@erc7579/lib/ExecutionLib.sol";
+
+import { Execution, Caveat, Delegation, ModeCode } from "../../src/utils/Types.sol";
+import { Counter } from "../utils/Counter.t.sol";
+import { CaveatEnforcerBaseTest } from "./CaveatEnforcerBaseTest.t.sol";
+import { AllowedMethodsEnforcer } from "../../src/enforcers/AllowedMethodsEnforcer.sol";
+import { IDelegationManager } from "../../src/interfaces/IDelegationManager.sol";
```
Comment (hanzel98) [inline .cursorrules:72]: This could be removed

File: .cursorrules [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+import { EncoderLib } from "../../src/libraries/EncoderLib.sol";
```
Comment (hanzel98) [inline .cursorrules:73]: These could be removed

**Review by dylandesrosier — APPROVED** [review]

**Review by hanzel98 — APPROVED** [review]

**Pull request merged**